mycore-base/src/main/java/org/mycore/datamodel/common/MCRXMLMetadataManager.java

```diff
@@ -18,379 +18,289 @@
 
 /**
- * Manages persistence of MCRObject and MCRDerivate xml metadata.
- * Provides methods to create, retrieve, update and delete object metadata
- * using IFS2 MCRMetadataStore instances.
+ * Provides an adapter to communicate with the configured {@link MCRXMLMetadataManagerAdapter} implementation.
  *
- * For configuration, at least the following properties must be set:
- *
- * MCR.Metadata.Store.BaseDir=/path/to/metadata/dir
- * MCR.Metadata.Store.SVNBase=file:///path/to/local/svndir/
- *
- * Both directories will be created if they do not exist yet.
- * For each project and type, a subdirectory will be created,
- * for example %MCR.Metadata.Store.BaseDir%/DocPortal/document/.
- *
- * The default IFS2 store is MCRVersioningMetadataStore, which
- * versions metadata using SVN in local repositories below SVNBase.
- * If you do not want versioning and would like to have better
- * performance, change the following property to
- *
- * MCR.Metadata.Store.DefaultClass=org.mycore.datamodel.ifs2.MCRMetadataStore
- *
- * It is also possible to change individual properties per project and object type
- * and overwrite the defaults, for example
- *
- * MCR.IFS2.Store.Class=org.mycore.datamodel.ifs2.MCRVersioningMetadataStore
- * MCR.IFS2.Store.SVNRepositoryURL=file:///use/other/location/for/document/versions/
- * MCR.IFS2.Store.SlotLayout=2-2-2-2
- *
- * See documentation of MCRStore and MCRMetadataStore for details.
- *
- * @author Frank Lützenkirchen
- * @author Jens Kupferschmidt
- * @author Thomas Scheffler (yagee)
+ * @author Christoph Neidahl (OPNA2608)
  */
 public class MCRXMLMetadataManager {
 
-    private static final Logger LOGGER = LogManager.getLogger();
-
-    /** The singleton */
+    /**
+     * Our own singleton.
+     */
     private static MCRXMLMetadataManager SINGLETON;
 
-    private HashSet<String> createdStores;
-
     /**
-     * The default IFS2 Metadata store class to use, set by MCR.Metadata.Store.DefaultClass
+     * The implementation's singleton.
      */
-    private Class defaultClass;
+    private static MCRXMLMetadataManagerAdapter IMPLEMENTATION;
 
     /**
-     * The default subdirectory slot layout for IFS2 metadata store, is 4-2-2 for 8-digit IDs,
-     * that means DocPortal_document_0000001 will be stored in the file
-     * DocPortal/document/0000/00/DocPortal_document_00000001.xml
+     * Reads the MCR.Metadata.Manager.Class to instantiate and return the configured xml metadata manager.
+     * If MCR.Metadata.Manager.Class is not set, an instance of {@link MCRDefaultXMLMetadataManager} is returned.
+     *
+     * @return an instance of the configured xml metadata manager if any is set, or MCRDefaultXMLMetadataManager
+     */
+    public static synchronized MCRXMLMetadataManager instance() {
+        if (SINGLETON == null) {
+            SINGLETON = new MCRXMLMetadataManager();
+        }
+        if (IMPLEMENTATION == null) {
+            IMPLEMENTATION = MCRConfiguration2
```

Comment:
The Property MCR.Metadata.Manager.Class should be defined in the mycore.properties.
We have the Rule that every default property should be in the mycore.properties, except for dynamic named properties.